framework/source/class/qx/event/Utils.js

```diff
@@ -67,7 +68,7 @@ qx.Class.define("qx.event.Utils", {
      */
     track: qx.core.Environment.select("qx.promise", {
       "true": function(tracker, fn) {
-        if (typeof fn !== "function" && !(fn instanceof qx.Promise)) {
+        if (typeof fn !== "function" && !(fn instanceof qx.Promise) && !(fn instanceof Promise)) {
```

Comment:
IMHO that kind of low level negotiation/detection of type is best suited to a common utility method, please can we move the implementation of this test into (for example) `qx.lang.Type.isPromise()`?  

Technically the Promises/A+ spec says that _“promise” is an object or function with a then method whose behavior conforms to this specification_, so in theory all an event has to do is `return { then() { console.log("hello"); } }`; the spec is at https://promisesaplus.com/

Reply:
Good idea. 👍